include/deal.II/lac/sparsity_pattern.h

```diff
@@ -1398,6 +1567,39 @@ SparsityPattern::n_nonzero_elements() const
 
 
 
+template <class Archive>
+inline void
+SparsityPatternBase::save(Archive &ar, const unsigned int) const
+{
+  // forward to serialization function in the base class.
+  ar &static_cast<const Subscriptor &>(*this);
+
+  ar &max_dim &rows &cols &max_vec_len &max_row_length &compressed;
+
+  ar &boost::serialization::make_array(rowstart.get(), max_dim + 1);
+  ar &boost::serialization::make_array(colnums.get(), max_vec_len);
+}
+
+
+
+template <class Archive>
+inline void
+SparsityPatternBase::load(Archive &ar, const unsigned int)
+{
+  // forward to serialization function in the base class.
+  ar &static_cast<Subscriptor &>(*this);
```

Comment:
the annoying thing I don't understand is that I was not able to use the parent class `save/load` in the derived as
```
ar &static_cast<const SparsityPatternBase &>(*this);
ar &store_diagonal_first_in_row;
```

I would expect this to work without any issues, unless there is something related to `boost::serialization::make_array` that prevents serializing scalars after it...

Reply:
ok, interesting and wierd part is that base class serialization is never called on the line above:
```
ar &static_cast<const SparsityPatternBase &>(*this);
```

neither is it called if I try to do it the recommended way:
```
ar &boost::serialization::base_object<const SparsityPatternBase>(*this);
```

😲 

will keep looking...

Reply:
should be solved now.